index.js

```diff
@@ -376,6 +378,12 @@ module.exports = class NoiseSecretStream extends Duplex {
     this.remotePublicKey = remotePublicKey
     this.handshakeHash = handshakeHash
 
+    // initialize secretbox for unordered messages
+    this._boxSecret = b4a.allocUnsafe(32)
```

Comment:
One tiny nit, it's easier to review crypto if all the sodium calls are grouped like so:
```
this._boxSecret = b4a.alloc(..)
this._boxNonce = b4a.alloc(..)

sodium.crypto_generichash(this._boxSecret...)
sodium.randombytes_buf(this._boxNonce)
```